### PR TITLE
Fix vcpkg-tool-meson clobbering the tools directory.

### LINKED ISSUE #37513: failed to compress folder
Screenshot 2024-03-17 070031

it comes with this warning will this create some issues later or not??

### PR DESCRIPTION
**Pull request by BillyONeal**

### CODE REVIEW

File: ports/vcpkg-tool-meson/vcpkg-port-config.cmake [hunk 1]
```diff
@@ -41,22 +41,25 @@ if(NOT SCRIPT_MESON)
         FILENAME "${download_filename}"
 
     )
-    file(REMOVE_RECURSE "${path_to_search}-tmp/../")
```
Comment (BillyONeal) [inline ports/vcpkg-tool-meson/vcpkg-port-config.cmake:44]: `${path_to_search}-tmp/../` is `${DOWNLOADS}/tools`

**Comment by dg0yt** [comment]
Maybe fixes https://github.com/microsoft/vcpkg/issues/36773?

**Comment by Neumann-A** [comment]
The reason it says `file(MAKE_DIRECTORY "${path_to_search}-tmp/../")` was because i observed `downloads/tools/meson/meson-sha512` but if that is no longer the case due to other changes within the port the extra `../` is of course wrong. I didn't actively notice the 7zip problems in my overlay install (maybe I just was lucky and something did a `vcpkg_find_acquire_program(7Z)` or I just did too many sequenced `vcpkg installs` or `--editable` tests) but an easy way to reproduce it is to `vcpkg install pkgconf` (or any meson based port). If CMake is provided by vcpkg it will just delete everything above that cmake folder. If a viable version of cmake is provided then the whole tools folder gets nuked. 

But another interesting questions arises here: How to detect and protect for such changes in the future? Do the vcpkg-tool tools need a different folder? Script tools would generally just get re-extracted.

**Review by Neumann-A — APPROVED** [review]

**Pull request merged**